src/ecc_compact/mod.rs

```diff
@@ -59,8 +59,13 @@ impl keypair::Sign for Keypair {
 impl TryFrom<&[u8]> for Keypair {
     type Error = Error;
     fn try_from(input: &[u8]) -> Result<Self> {
-        let network = Network::try_from(input[0])?;
-        let secret = p256::SecretKey::from_bytes(&input[1..])?;
+        let (network_byte, key_bytes) = {
+            let (nb, kbs) = input.split_at(1);
```

Comment:
why do this split at at all? Is it because you specifically want the last KEYPAIR_LENGTH of bytes? 

Why not  to `[1..usize::min(kbs.len(), KEYPAIR_LENGTH - 1)]`  instead?

Reply:
right but there's still some right? So this adds some over head for marginal upside? Either way not a biggie, just triggered on the apparent additional complexity

Reply:
I consider this less.... cogitively complex? But you have a valid point.

Reply:
> Why not to [1..usize::min(input.len() - 1, KEYPAIR_LENGTH - 1)] on line 63 instead?

FYI: that has an indexing error in it.